kubernetes/src/main/java/com/linecorp/armeria/client/kubernetes/ArmeriaHttpClientFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.kubernetes;
+
+import com.linecorp.armeria.client.WebClientBuilder;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.http.HttpClient;
+import io.fabric8.kubernetes.client.http.HttpClient.Factory;
+
+/**
+ * An {@link HttpClient.Factory} for creating {@link ArmeriaHttpClient}.
+ */
+public class ArmeriaHttpClientFactory implements HttpClient.Factory {
+    @Override
+    public ArmeriaHttpClientBuilder newBuilder() {
```

Comment:
Not now. I followed the upstream code pattern.
https://github.com/fabric8io/kubernetes-client/blob/56a6c2c4f336cc6f64c19029a55c2d3d0289344f/httpclient-vertx/src/main/java/io/fabric8/kubernetes/client/vertx/VertxHttpClientFactory.java#L36-L37
However, I prefer keeping `ArmeriaHttpClientBuilder` as the return type because we may add Armeria-specific options to the builder class later.

Reply:
I see, I thought this Factory would mainly be used via SPI and didn't realize users could also instantiate the builder directly. I guess it's fine to leave as is then (It might help adding a description since there is no context for reviewers 😅 )

Reply:
I believe you convinced me. 😉
Users can fully customize the underlying Armeria `WebClient` with  `ArmeriaHttpClientFactory.additionalConfig(WebClientBuilder)`.
A custom client factory can be set via https://github.com/fabric8io/kubernetes-client/blob/e7f734bd5922bce0a78e9babbbdfce4d564560bb/kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/KubernetesClientBuilder.java#L117